### PR TITLE
[RHPAM-4898] Fix concurrency issue in KafkaServerRegistration

### PR DESCRIPTION
**Pull request by martinweiler**
Related JIRA: https://issues.redhat.com/browse/RHPAM-4898

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java [hunk 1]
```diff
@@ -42,26 +43,26 @@
 
 class KafkaServerRegistration {
     
-    private Map<String, Map<SignalDesc, Map<DeploymentId, SortedSet<VersionedDeploymentId>>>> topic2Signal = new HashMap<>();
-    private Map<String, Map<MessageDesc, Map<DeploymentId, SortedSet<VersionedDeploymentId>>>> topic2Message = new HashMap<>();
+    private Map<String, Map<SignalDesc, Map<DeploymentId, SortedSet<VersionedDeploymentId>>>> topic2Signal = new ConcurrentHashMap<>();
+    private Map<String, Map<MessageDesc, Map<DeploymentId, SortedSet<VersionedDeploymentId>>>> topic2Message = new ConcurrentHashMap<>();
 
-    synchronized void close() {
+    void close() {
```
Comment (elguardian) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java:49]: do not remove

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java [hunk 2]
```diff
@@ -42,26 +43,26 @@
...
         topic2Signal.clear();
         topic2Message.clear();
     }
 
-    synchronized boolean isEmpty() {
+    boolean isEmpty() {
```
Comment (elguardian) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java:54]: do not remove

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java [hunk 3]
```diff
@@ -42,26 +43,26 @@
...
         return topic2Signal.isEmpty() && topic2Message.isEmpty();
     }
 
-    synchronized Set<String> addRegistration(DeploymentEvent event) {
+    Set<String> addRegistration(DeploymentEvent event) {
```
Comment (elguardian) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java:58]: do not remove

File: kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java [hunk 4]
```diff
@@ -42,26 +43,26 @@
...
         for (DeployedAsset asset : event.getDeployedUnit().getDeployedAssets()) {
             updateTopics(new DeploymentIdFactory(event), (ProcessDefinition) asset);
         }
         return getTopicsRegistered();
     }
 
-    synchronized Set<String> removeRegistration(DeploymentEvent event, Consumer<String> topicProcessed) {
+    Set<String> removeRegistration(DeploymentEvent event, Consumer<String> topicProcessed) {
```
Comment (elguardian) [inline kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java:65]: do not remove

**Review by elguardian — CHANGES_REQUESTED** [review]

**Comment by martinweiler** [comment]
Thanks @elguardian - please review now.

**Review by elguardian — APPROVED** [review]

**Comment by mareknovotny** [comment]
> Jenkins retest this please

Jenkins is not triggered on all branches, just selected maintained ones. so this is useless, you can see only GHA CI

**Comment by elguardian** [comment]
@martinweiler we need this against main

**Pull request merged**